app/models/bulk_download.rb

```diff
@@ -23,6 +23,47 @@ class BulkDownload < ApplicationRecord
   attr_accessor :params
 
   validates :status, presence: true, inclusion: { in: [STATUS_WAITING, STATUS_RUNNING, STATUS_ERROR, STATUS_SUCCESS] }
+  validate :params_checks
+
+  def params_checks
+    set_params
+    params = self.params
+
+    params_value = lambda do |key|
```

Comment:
Isn't this just a `dig` (https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig)?
i.e. `params.dig(key, "value") ?

Reply:
It was a little trickier when it could be an `ActionController::Parameter`. I now convert from ACP to a Hash in the controller, so it's guaranteed to be a hash here. You're right, this method is no longer needed.